### ISSUE TITLE
cscope-index-files doesn't work with paths using whitespaces

### ISSUE THREAD

**Issue by andijh92** [dkogan/xcscope.el#14]
Hi,

cscope-index-files doesn't work with paths using whitespaces. They should be enclosed in double quotes.

Regards,
Andy

cscope version: 20160628.2324

**Comment by dkogan** [comment]
andijh92 <notifications@github.com> writes:

> cscope-index-files doesn't work with paths using whitespaces. They
> should be enclosed in double quotes.

Hi. Sorry it took so long to reply. I was under the impression that
files with spaces in their names weren't supported by cscope at all (the
C tool that xcscope.el relies upon). Is this not the case? If you
enclose the filenames in double quotes then you're able to index and
search these files?

**Comment by bertogg** [comment]
Hi, from the cscope(1) manual page:

```
Filenames  in the namefile that contain whitespace have to be
enclosed in "double quotes".   Inside  such  quoted  filenames,
any double-quote and backslash characters have to be escaped
by backslashes.
```

(this is from cscope version 15.9)

I tried this quick change to xcscope.el and I confirm that it works:

```
--- xcscope.el.orig     2020-08-27 12:59:35.756439407 +0200
+++ xcscope.el  2020-08-27 12:59:42.340456818 +0200
@@ -2736,7 +2736,7 @@
                     "(" "-type" "f" "-o" "-type" "l" ")"
 
                     ;; if we made it here, take the result
-                    "-print"))))
+                    "-printf" "\"%p\"\n"))))
 
     (let ((findcmds
            (concat "echo 'Creating list of files to index ...'\n"

```

This also removes all the annoying "cannot find file ..." that I was getting all the time.

Problems:

- This works for white spaces but if the file name contains other special characters they would need to be escaped. I don't care about e.g double quotes or backslashes on file names (I think they are extremely rare), but white spaces are relatively common.
- xcscope still cannot parse the output so if I use `cscope-find-global-definition` I get the results on the Emacs buffer but I cannot click on them ("File does not have expected format"). In my particular case I don't care so much about those files so my goal is simply to remove the "cannot find file" warnings.

**Comment by dkogan** [comment]
Hi. Thanks for the report. To clarify, the proposed change makes
whitespace-in-filenames work for you, but other crazy characters remain
unescaped and broken? Do you see any downsides to that modification?

**Comment by bertogg** [comment]
I was having a closer look at this and I created a new patch that hopefully quotes all filenames correctly.

I gave it a try with files named    `one two.c`  ,  `one"two"three.c`  and `one\two.c`     and it seems to work fine now.

Now the index is generated correctly. If you search a symbol that is inside a file with a whitespace in the name cscope still complains a bit and does not work 100%, but at least it does work partially (that's a cscope bug, not a problem with the Emacs elisp file). Files with double quotes or backslashes seem to work fine.

See https://github.com/dkogan/xcscope.el/pull/23 for more details.

And here are related discussion for cscope: https://sourceforge.net/p/cscope/bugs/282/ and https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=579531
